### ISSUE TITLE
How to calculate the loss of verification set

### ISSUE THREAD

**Issue by BelieferQAQ** [Project-MONAI/tutorials#801]
def validation(val_loader,epoch,max_epochs):
    model.eval()
    start_time = time.time()
    val_loss = 0
    with torch.no_grad():
        for step, batch in enumerate(val_loader):
            val_inputs, val_labels = (batch["image"].cuda(), batch["label"].cuda())
            print('the label shape:{}'.format(val_labels.shape))

            val_outputs = sliding_window_inference(val_inputs, (96, 96, 96), 4, model)
            
_###             **val_loss_batch = loss_function(val_inputs,val_outputs)
            val_loss += val_loss_batch.item()**_

            print('the output shape:{}'.format(val_outputs.shape))
            val_labels_list = decollate_batch(val_labels)
            val_labels_convert = [
                post_label(val_label_tensor) for val_label_tensor in val_labels_list
            ]
            val_outputs_list = decollate_batch(val_outputs)
            val_output_convert = [
                post_pred(val_pred_tensor) for val_pred_tensor in val_outputs_list
            ]
            print('the laellist shape:{}'.format(val_labels_convert.shape))
            print('the outputlist shape:{}'.format(val_output_convert.shape))
            dice_metric(y_pred=val_output_convert, y=val_labels_convert)
        val_loss = val_loss/(step+1)
        mean_dice_val = dice_metric.aggregate().item()
        dice_metric.reset()
        print("Val {}/{} {}/{}".format(epoch,max_epochs, step, len(val_loader)),
              "loss: {:.4f}".format(val_loss),
              "dice",mean_dice_val,
              "time {:.2f}s".format(time.time() - start_time)
             )
    return val_loss

If I correct it like this, I will report an error. The input and output dimensions are inconsistent. How can I correct it？thank you

**Comment by BelieferQAQ** [comment]
model = SwinUNETR(
    img_size=(96, 96, 96),
    in_channels=1,
    out_channels=14,
    feature_size=48,
    use_checkpoint=True,
)

torch.backends.cudnn.benchmark = True
loss_function = DiceCELoss(to_onehot_y=True, softmax=True)
optimizer = torch.optim.AdamW(model.parameters(), lr=1e-4, weight_decay=1e-5)

**Comment by KumoLiu** [comment]
Hi @BelieferQAQ,
From the code you posted, it looks like you have some problems when you use `sliding_window_inference`. 
The input and output sizes of `sliding_window_inference` are kept consistent and do not need to be post-processed dimensionally. You may find some guidance in the 3d_segentation tutorials.  Thanks!

**Comment by BelieferQAQ** [comment]
![image](https://user-images.githubusercontent.com/91660759/179472229-fb603370-d3eb-46a9-a295-9683c60309c1.png)
![image](https://user-images.githubusercontent.com/91660759/179472276-6efa82ee-6f4d-43f0-bc77-9a2786a5ebad.png)
![image](https://user-images.githubusercontent.com/91660759/179472285-79d4e2fe-5398-43b2-b146-c5c439db6863.png)
This is problem of monai version？ my monai verison is 0.8

**Comment by KumoLiu** [comment]
Hi @BelieferQAQ,
The shape of val_outputs is B*C*W*H*D, you set your output channel as 14, but your labels channel is 1? Maybe you should check your data's channel?

**Comment by BelieferQAQ** [comment]
![image](https://user-images.githubusercontent.com/91660759/179510879-7738919d-0c54-4441-bed5-4d4bdd620089.png)
![image](https://user-images.githubusercontent.com/91660759/179510918-d0dd88cf-a81e-4bc5-8aad-21e8bf4bca7b.png)
Why in the training phase, label and logit_ The shape of map is different, but it can calculate loss？

**Comment by KumoLiu** [comment]
Hi @BelieferQAQ,
Change the order of the inputs, please. I hadn't seen that the loss you use has put label onehot, so the number of channels will not be a problem. Try like this:
`loss = loss_function(val_outputs, val_inputs)`
